### ISSUE TITLE
Merge xsim USB changes back to tools projects.

### ISSUE THREAD

**Issue by DaveAtkinsonXMOS** [xmos/lib_xud#147]
USB testing required changes to the xCore simulator and some associated utilities. I'm creating this issue to provide visibility of this work.

**Comment by DaveAtkinsonXMOS** [comment]
All known problems ironed out on pull request, waiting for tools team to perform final integration.

**Comment by DaveAtkinsonXMOS** [comment]
All outstanding pull requests for xsim USB functionality merged into main tools repos. Done with a capital D.

**Comment by DaveAtkinsonXMOS** [comment]
Oops. I closed this slightly prematurely. Although all the changes have been merged into the respective repositories, those commits have yet to be applied to the pull requests in the various 'combined' repos (tools_common, xsim_combined, xgdb_combined), which is what is required to build a version of tools containing the necessary changes via tools_installers/jenkins. Update to follow when that happens.
